framework/src/outputs/FileOutput.C

```diff
@@ -74,12 +77,24 @@ FileOutput::FileOutput(const InputParameters & parameters) :
     _file_base += buffer;
   }
 
-  // Check the file directory of file_base
+  // Check the file directory of file_base and create if needed
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
-
+  {
+    //Directory does not exist. Loop through incremental directories and create as needed.
+    size_t pos = 0;
+    std::string inc_path;
+    do
```

Comment:
No. Making tokenize a function that splits paths into components does not
seem like the right thing. There should be a separate function for this.
Or... boost:: filesystem (path iterators)... ;-)

On Fri, May 27, 2016, 8:20 PM Derek Gaston <notifications@github.com> wrote:

> In framework/src/outputs/FileOutput.C
> <https://github.com/idaholab/moose/pull/7101#discussion_r64982569>:
>
> >    std::string base = "./" + _file_base;
> >    base = base.substr(0, base.find_last_of('/'));
> >    if (access(base.c_str(), W_OK) == -1)
> > -    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
> > -
> > +  {
> > +    //Directory does not exist. Loop through incremental directories and create as needed.
> > +    size_t pos = 0;
> > +    std::string inc_path;
> > +    do
>
> I suspect that we have many places throughout the framework that don't
> deal with \/ properly (file / directory handling for restart raises its
> hand).
>
> This is actually another reason *TO* use tokenize()... because if we make
> it deal with escaped characters properly in the future then it will fix
> this too 😄
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7101/files/bd2be34efac386b494aa5f1183bfd7561098d68d#r64982569>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPt7W-35p1JgJcUC3hrz3fJGY3XYeks5qF6ZhgaJpZM4IoKUf>
> .
>